tic_tac_toe/game.py

```diff
@@ -87,18 +142,68 @@ def move(game, player, position):
     checks before the actual movement is done.
     After registering the movement it must check if the game is over.
     """
-    pass
-
+    # Check for winners or a full board
+    if game['winner'] or _board_is_full(game['board']):
+        raise InvalidMovement("Game is over.")
+        
+    if not _position_is_valid(position):
+        raise InvalidMovement('Position out of range.')
+        
+    # check if it is (player)s turn
+    if player != game['next_turn']:
+        if player == 'X':
+            tmp_player = 'O'
+        else:
+            tmp_player = 'X'
+        raise InvalidMovement('"{0}" moves next'.format(tmp_player))
+        
+    # check if the desired position is empty
+    if not _position_is_empty_in_board(position, game['board']):
+        raise InvalidMovement("Position already taken.")
+    
+    
+
+    # Register the move
+    game['board'][position[0]][position[1]] = player
+    
+    # Check if we have a winner
+    if _check_winning_combinations(game['board'], player):
+        game['winner'] = player
+        raise GameOver('"{0}" wins!'.format(player))
+    # Check if there is space left on the board 
+    if not _board_is_full(game['board']):
+        if game['next_turn'] == "X":
+            game['next_turn'] = "O"
+        else:
+            game['next_turn'] = "X"
+    else: # Board is full, Game has been tied
+        raise GameOver("Game is tied!")
 
 def get_board_as_string(game):
     """
     Returns a string representation of the game board in the current state.
     """
-    pass
+
+    board = """
+%s  |  %s  |  %s
+--------------
+%s  |  %s  |  %s
+--------------
+%s  |  %s  |  %s\n""" % (game['board'][0][0], game['board'][0][1], game['board'][0][2],
+             game['board'][1][0], game['board'][1][1], game['board'][1][2], game['board'][2][0],
+             game['board'][2][1], game['board'][2][2])
```

Comment:
Hi! I'm not entirely sure but instead of using % I think .format() would be better.